build.rs

```diff
@@ -56,17 +60,53 @@ fn feature_check(filename: &str) -> bool {
     check_build.file(path).try_compile("check_perf_pt").is_ok()
 }
 
+fn build_libipt() {
+    eprintln!("Building libipt...");
+    env::set_current_dir(&Path::new(C_DEPS_PATH)).unwrap();
+    let res = Command::new("make")
+        .arg("libipt")
+        .output()
+        .unwrap_or_else(|_| panic!("Spawning the libipt build failed"));
+    if !res.status.success() {
+        eprintln!("libipt build failed\n>>> stdout");
+        eprintln!("stdout: {}", String::from_utf8_lossy(&res.stdout));
+        eprintln!("\n>>> stderr");
+        eprintln!("stderr: {}", String::from_utf8_lossy(&res.stderr));
+        panic!();
+    }
+    env::set_current_dir(&Path::new("..")).unwrap();
+}
+
+
 fn main() {
     let mut c_build = gcc::Build::new();
 
     // Check for a new enough Perf to support Intel PT.
     #[cfg(target_os = "linux")] {
         if feature_check("check_perf_pt.c") {
             c_build.file("src/backends/perf_pt/perf_pt.c");
+            c_build.file("src/backends/perf_pt/decode.c");
+
+            // Decide whether to build our own libipt.
+            println!("cargo:rerun-if-env-changed=WITH_SYSTEM_IPT");
+            println!("cargo:rerun-changed={}", C_DEPS_PATH);
+            match env::var("WITH_SYSTEM_IPT") {
```

Comment:
Up to you, but this would be 1 line shorter, and 1 level of indentation shallower, if you used `if let` and `else`.